rfd/0062-tsh-proxy-template.md

```diff
@@ -126,11 +162,17 @@ is equivalent to the following when connecting to `node-1.leaf1.us.acme.com`:
 tsh proxy ssh -J leaf1.us.acme.com:3080 %r@node-1:3022
 ```
 
-### Auto-login
+#### Auto-login
 
 To further improve the UX, we should make sure that `tsh proxy ssh` command
 does not require users to log into each individual leaf cluster beforehand.
 
 Users should be able to login once to their root cluster and use that
 certificate to set up proxy with the extracted leaf's proxy (similar to proxy
 jump scenario).
+
+### Proxy Templates for `tsh ssh`
+
+Proxy Templates can also be used with `tsh ssh` and have feature parity with `ssh`.
```

Comment:
Will `tsh ssh` only take "cluster" into account, or "proxy" too?

Reply:
Same as below, `tsh ssh` and `tsh proxy ssh` will handle both `cluster` and `proxy` according to the specification of the proxy templates. However, a `proxy` entry will no longer be required for the proxy template to parse.

For example, a user could use the following template to just add a `.leaf1` to a node name to get `--cluster=leaf1`. Since `proxy` is not provided, it won't be changed with the template.

```yaml
proxy_templates:
- template: '^(\w*)\.(leaf1)$'
- cluster: '$2'
```

Reply:
Great, can you include this information in the design.